plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ScaleDeploymentHandler.java

```diff
@@ -49,37 +66,114 @@
 	
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IReplicationController rc = getSelectedElement(event, IReplicationController.class);
-		if(rc != null) {
-			scaleUsing(event, rc, rc.getName());
+		IServiceWrapper selected = getSelectedElement(event, IServiceWrapper.class);
+		if(selected == null) {
 			return null;
 		}
-		IServiceWrapper deployment = getSelectedElement(event, IServiceWrapper.class);
-		if(deployment != null) {
-			Collection<IResourceWrapper<?, ?>> rcs = deployment.getResourcesOfKind(ResourceKind.REPLICATION_CONTROLLER);
-			if(!rcs.isEmpty()) {
-				//there should be only 1 per deployment, we'll assume this is true
-				rc = (IReplicationController) rcs.iterator().next().getWrapped();
-				scaleUsing(event, rc, deployment.getWrapped().getName());
+
+		String name = null; //Show in job name and dialog title.
+		IServiceWrapper deployment = null;
+		IReplicationController rc = null;
+
+		if(selected instanceof IServiceWrapper) {
```

Comment:
No but you don't need a cast as variable is of type IServiceWrapper and may not be null at this point